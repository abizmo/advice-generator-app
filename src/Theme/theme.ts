const theme = {
  borderRadius: {
    500: '.25rem',
    700: '.75rem',
    round: '50%',
  },
  palette: {
    common: {
      500: 'hsl(217, 19%, 38%)',
      700: 'hsl(217, 19%, 24%)',
      900: 'hsl(218, 23%, 16%)',
    },
    primary: {
      contrast: 'hsl(193, 38%, 86%)',
      main: 'hsl(150, 100%, 66%)',
    },
  },
  sizes: {
    1: '.0625rem',
    2: '.125rem',
    4: '.25rem',
    8: '.5rem',
    12: '.75rem',
    16: '1rem',
    24: '1.5rem',
    32: '2rem',
    40: '2.5rem',
    48: '3rem',
    56: '3.5rem',
    64: '4rem',
  },
};

export default theme;
