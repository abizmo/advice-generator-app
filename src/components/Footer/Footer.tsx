import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.footer`
  font-size: ${({ theme }) => theme.fontSize[100]};
  font-weight: 400;
  text-align: center;

  & a {
    color: ${({ theme }) => theme.palette.primary.main};
    font-weight: 700;
    text-decoration: none;
  }
`;

function Footer() {
  return (
    <Wrapper>
      Challenge by
      {' '}
      <a href="https://www.frontendmentor.io?ref=challenge" target="_blank" rel="noreferrer">Frontend Mentor</a>
      .
      Coded by
      {' '}
      <a href="https://www.abizmo.dev">Abi√°n Izquierdo</a>
      .
    </Wrapper>
  );
}

export default Footer;
