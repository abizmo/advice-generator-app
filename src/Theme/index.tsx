import React from 'react';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import theme from './theme';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: ${theme.palette.common[900]};
    color: ${theme.palette.primary.contrast};
    font-family: 'Manrope', sans-serif;
    font-weight: 700;
  }
`;

// eslint-disable-next-line react/prop-types
function Theme({ children }: any): JSX.Element {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
}

export default Theme;
