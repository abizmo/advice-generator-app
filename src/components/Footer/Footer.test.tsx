import React from 'react';
import { render } from '@testing-library/react';

import Footer from './Footer';
import Theme from '../../Theme';

test('should render and MatchSnapshot', () => {
  const { asFragment } = render(
    <Theme>
      <Footer />
    </Theme>,
  );

  expect(asFragment()).toMatchSnapshot();
});
