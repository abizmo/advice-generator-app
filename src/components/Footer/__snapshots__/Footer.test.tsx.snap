// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and MatchSnapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-bczRLJ dWMZnY"
  >
    Challenge by 
    <a
      href="https://www.frontendmentor.io?ref=challenge"
      rel="noreferrer"
      target="_blank"
    >
      Frontend Mentor
    </a>
    . Coded by 
    <a
      href="https://www.abizmo.dev"
    >
      Abi√°n Izquierdo
    </a>
    .
  </footer>
</DocumentFragment>
`;
